### DISCUSSION TITLE
[v6] Roadmap?

### DISCUSSION THREAD

**Discussion by brianespinosa** [remix-run/react-router#8571] [General]
I spent a while deep searching issues and discussions, as well as the updated docs site to find out if there was mention anywhere of a roadmap for a lot of the features that have been discussed for v6. I appreciate that we have a full release now, and I understand the decision to hold some of the more half-baked features back until their APIs can be more solidified.

I think it would be helpful for a lot of the teams relying on RR in general to see what the roadmap is for some of these features. Many of them I have seen others state that these are things blocking them from moving forward off of an older beta version of v6 or from v5. These are all features it seems that core maintainers have stated will be in RRv6, but with statements like "the near future" which is relative.

Does a roadmap of priorities exist somewhere that I have missed in all of my searching? Even something with a list of these things in rank priority order would be extremely helpful. Dates or quarter estimates would be great, but it would be better to see what the team thinks is the most important thing to get out the door next.

- [`usePrompt`, `useBlocker`](https://github.com/remix-run/react-router/issues/8139#issuecomment-954425560)
- [Animation](https://github.com/remix-run/react-router/discussions/8008)
- [Absolute paths in descendant `<Routes />` with `basepath`](https://github.com/remix-run/react-router/issues/8035#issuecomment-962632573)

I know there are other items people are looking for as well, but these are the three that I have seen which seem to have received the endorsement of core maintainers.

_Edit: To be clear... this is not a message asking when core maintainers are going to ship this. RR is open source. If there is a roadmap with some good broad stroke requirements written up, I am sure we could empower the community using RR to step up and take a stab at getting some of these things working to a point we can kick the tires a bit and make some decisions about getting these features in._

**Comment by ryanflorence** [comment]
Hey Brian 👋

We're still getting Remix off the ground, both technically and organizationally. While this kind of stuff is absolutely something we want to be able to communicate better, we're not that organized yet 😅

Our current roadmap includes the following:

- Backwards Compatibility layer to be able to upgrade from v5 to v6 incrementally and without downtime is coming ASAP (just docs and the actual release left to do)
- [Remixing React Router](https://remix.run/blog/remixing-react-router) is underway, should have experimental releases starting in a week or two.
- Merging the Remix + Router repos is happening within the next month

Once those things are done we'll be figuring out the next things we'll be working on in React Router. The "Remixing React Router" is a decent change to the internals. Once that's done we can revisit the items you've listed here.

Hope that helps!

**Reply by brianespinosa** [reply]
Hi Ryan,

It is incredibly helpful to know where these things stand in your vision, even if they are not in an actual roadmap yet. Thank you so much for sharing this info!

**Comment by abenhamdine** [comment]
Roadmap is now public : https://github.com/orgs/remix-run/projects/5/views/1  

see also this post https://remix.run/blog/open-development

**Reply by machour** [reply]
That's remix's roadmap, not react-router.

**Reply by abenhamdine** [reply]
oups sorry, looks I messed up when googling for the roadmap !
Anyway I let my comment here for future readers given that future of react-router and remix are somehow tied.
